### PR TITLE
[SystemChrome] setPreferredOrientations implementation added

### PR DESCRIPTION
**Pull request by pkosko**
[Before] not implemented

[After] Using API below allows to narrow some rotations of application UI
e.g.
/// to limit some orientations
List<DeviceOrientation> orientations = [DeviceOrientation.portraitUp, DeviceOrientation.landscapeLeft];
SystemChrome.setPreferredOrientations(orientations).then(
    (value) => print("preferred orientations were set"));

/// to clear previous limitations
List<DeviceOrientation> orientations = [];
SystemChrome.setPreferredOrientations(orientations).then(
    (value) => print("preferred orientations were CLEARED"));

This PR is related to part of https://github.sec.samsung.net/f-project/home/issues/35:

Even though there's no corresponding Tizen system API for locking the app orientation, can't we somehow manage the desired orientation information in the embedder to use with TizenEmbedderEngine::SetWindowOrientation? I think we should investigate this particular method in more details because many apps may want to use a fixed orientation.

### CODE REVIEW

File: shell/platform/tizen/tizen_renderer.h [hunk 1]
```diff
@@ -36,6 +37,7 @@ class TizenRenderer {
   virtual void SetRotate(int angle) = 0;
   virtual void ResizeWithRotation(int32_t x, int32_t y, int32_t width,
                                   int32_t height, int32_t degree) = 0;
+  virtual void SetPreferredOrientations(std::vector<int> rotations) = 0;
```
Comment (pwasowski2) [inline shell/platform/tizen/tizen_renderer.h:40]:
```suggestion
  virtual void SetPreferredOrientations(const std::vector<int>& rotations) = 0;
```

If you agree to change this, please remember to change signatures of definitions of this method.

File: shell/platform/tizen/tizen_renderer_evas_gl.cc [hunk 2]
```diff
@@ -709,3 +709,8 @@ void TizenRendererEvasGL::ResizeWithRotation(int32_t x, int32_t y,
 void TizenRendererEvasGL::SendRotationChangeDone() {
   elm_win_wm_rotation_manual_rotation_done(evas_window_);
 }
+
+void TizenRendererEvasGL::SetPreferredOrientations(std::vector<int> rotations) {
+  elm_win_wm_rotation_available_rotations_set(
+      evas_window_, (const int*)(rotations.data()), rotations.size());
```
Comment (pwasowski2) [inline shell/platform/tizen/tizen_renderer_evas_gl.cc:715]:
```suggestion
      evas_window_, static_cast<const int*>(rotations.data()), rotations.size());
```

File: shell/platform/tizen/channels/platform_channel.cc [hunk 3]
```diff
@@ -43,7 +48,32 @@ void PlatformChannel::HandleMethodCall(
...
+    const std::string kPortraitUp = "DeviceOrientation.portraitUp";
+    const std::string kPortraitDown = "DeviceOrientation.portraitDown";
+    const std::string kLandscapeLeft = "DeviceOrientation.landscapeLeft";
+    const std::string kLandscapeRight = "DeviceOrientation.landscapeRight";
+    const std::map<std::string, int> orientation_mapping = {
+        {kPortraitUp, 0},
+        {kLandscapeLeft, 90},
+        {kPortraitDown, 180},
+        {kLandscapeRight, 270},
+    };
```
Comment (pwasowski2) [inline shell/platform/tizen/channels/platform_channel.cc:60]:
```suggestion
    static const std::string kPortraitUp = "DeviceOrientation.portraitUp";
    static const std::string kPortraitDown = "DeviceOrientation.portraitDown";
    static const std::string kLandscapeLeft = "DeviceOrientation.landscapeLeft";
    static const std::string kLandscapeRight = "DeviceOrientation.landscapeRight";
    static const std::map<std::string, int> orientation_mapping = {
        {kPortraitUp, 0},
        {kLandscapeLeft, 90},
        {kPortraitDown, 180},
        {kLandscapeRight, 270},
    };
```

They'd be initialized only once this way.

File: shell/platform/tizen/channels/platform_channel.cc [hunk 4]
```diff
@@ -43,7 +48,32 @@ void PlatformChannel::HandleMethodCall(
...
+
+    const auto& list = call.arguments()[0];
+    std::vector<int> rotations;
+    for (rapidjson::Value::ConstValueIterator itr = list.Begin();
```
Comment (pwasowski2) [inline shell/platform/tizen/channels/platform_channel.cc:64]: Consider logging values passed by the user with `FT_LOGD` to make debugging easier.

**Comment by pkosko** [comment]
@pwasowski2 Thank you for you feedback. I've applied all your suggestions.

**Review by pwasowski2 — APPROVED** [review]
LGTM 👏

File: shell/platform/tizen/channels/platform_channel.h [hunk 5]
```diff
@@ -7,15 +7,18 @@
...
 class PlatformChannel {
  public:
-  explicit PlatformChannel(flutter::BinaryMessenger* messenger);
+  explicit PlatformChannel(flutter::BinaryMessenger* messenger,
+                           TizenRenderer* renderer);
   virtual ~PlatformChannel();
 
  private:
   std::unique_ptr<flutter::MethodChannel<rapidjson::Document>> channel_;
+  TizenRenderer* tizen_renderer;
```
Comment (swift-kim) [inline shell/platform/tizen/channels/platform_channel.h:21]: nit: naming convention

```suggestion
  TizenRenderer* tizen_renderer_ = nullptr;
```

**Review by swift-kim — APPROVED** [review]

**Comment by pkosko** [comment]
@swift-kim I've renamed member name to tizen_renderer_ as you suggested.

**Pull request merged**